SpokestackTests/TFLiteWakewordRecognizerTest.swift

```diff
@@ -0,0 +1,234 @@
+    
+    func hexStringToData(hexString: String) -> Data? {
+        let len = hexString.count / 2
+        var data = Data(capacity: len)
+        for i in 0..<len {
+            let j = hexString.index(hexString.startIndex, offsetBy: i*2)
+            let k = hexString.index(j, offsetBy: 2)
+            let bytes = hexString[j..<k]
+            if var num = UInt8(bytes, radix: 16) {
+                data.append(&num, count: 1)
+            } else {
+                return nil
+            }
+        }
+        return data
+    }
+    
+    override func setUp() {
+        //let filterHexString = filter.map { String(format: "%02hhx", $0) }.joined()
+        //let filterData = hexStringToData(hexString: MockWakewordModels.filterString)
+        //let _ = FileManager.default.createFile(atPath: MockWakewordModels.filterPath, contents: filterData, attributes: .none)
+    }
+    
+    func testInvoke() {
+        // filter
+        let filter = try! Interpreter(modelPath: MockWakewordModels.filterPath)
+        try! filter.allocateTensors()
+        _ = MockWakewordModels.filterInput.withUnsafeBytes { try! filter.copy(Data($0), toInputAt: 0) }
+        _ = try! filter.invoke()
+        let filterOutput = try! filter.output(at: MockWakewordModels.validIndex)
+        XCTAssertEqual(filterOutput.data, MockWakewordModels.filterOutputData)
+        let filterResult = filterOutput.data.toArray(type: Int32.self, count: filterOutput.data.count/4)
+        XCTAssertEqual(filterResult, MockWakewordModels.filterOutput)
+        
+        // encode
+        let encoder = try! Interpreter(modelPath: MockWakewordModels.encodePath)
+        try! encoder.allocateTensors()
+        _ = MockWakewordModels.encodeInput.withUnsafeBytes { try! encoder.copy(Data($0), toInputAt: 0) }
+        _ = MockWakewordModels.input.withUnsafeBytes { try! encoder.copy(Data($0), toInputAt: 1) }
+        _ = try! encoder.invoke()
+        let encoderOutput = try! encoder.output(at: MockWakewordModels.validIndex)
+        XCTAssertEqual(encoderOutput.data, MockWakewordModels.encodeOutputData)
+        let encoderResult = encoderOutput.data.toArray(type: Int32.self, count: encoderOutput.data.count/4)
+        XCTAssertEqual(encoderResult, MockWakewordModels.encodeOutput)
+        
+        // detect
+        let detect = try! Interpreter(modelPath: MockWakewordModels.detectPath)
+        try! detect.allocateTensors()
+        _ = MockWakewordModels.detectInput.withUnsafeBytes { try! detect.copy(Data($0), toInputAt: 0) }
+        _ = try! detect.invoke()
+        let detectOutput = try! detect.output(at: MockWakewordModels.validIndex)
+        XCTAssertEqual(detectOutput.data, MockWakewordModels.detectOutputData)
+        let detectResult = detectOutput.data.toArray(type: Int32.self, count: detectOutput.data.count/4)
+        XCTAssertEqual(detectResult, MockWakewordModels.detectOutput)
+    }
+    
+    func testStartStop() {
+        // setup
```

Comment:
As verbose and identical as the setup is for the next 3 tests, it'd be a good candidate for a separate function.